templates/tls-init-job.yaml

```diff
@@ -61,27 +61,11 @@ spec:
             - "/bin/sh"
             - "-ec"
             - |
-              {{- if (not (and .Values.global.tls.caCert.secretName .Values.global.tls.caKey.secretName)) }}
-              consul tls ca create \
-                -domain={{ .Values.global.domain }}
-              curl -s -X POST --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
-                https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/api/v1/namespaces/${NAMESPACE}/secrets \
-                -H "Authorization: Bearer $( cat /var/run/secrets/kubernetes.io/serviceaccount/token )" \
-                -H "Content-Type: application/json" \
-                -H "Accept: application/json" \
-                -d "{ \"kind\": \"Secret\", \"apiVersion\": \"v1\", \"metadata\": { \"name\": \"{{ template "consul.fullname" . }}-ca-cert\", \"namespace\": \"${NAMESPACE}\" }, \"type\": \"Opaque\", \"data\": { \"tls.crt\": \"$( cat {{ .Values.global.domain }}-agent-ca.pem | base64 | tr -d '\n' )\" }}" > /dev/null
-              curl -s -X POST --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
-                https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/api/v1/namespaces/${NAMESPACE}/secrets \
-                -H "Authorization: Bearer $( cat /var/run/secrets/kubernetes.io/serviceaccount/token )" \
-                -H "Content-Type: application/json" \
-                -H "Accept: application/json" \
-                -d "{ \"kind\": \"Secret\", \"apiVersion\": \"v1\", \"metadata\": { \"name\": \"{{ template "consul.fullname" . }}-ca-key\", \"namespace\": \"${NAMESPACE}\" }, \"type\": \"Opaque\", \"data\": { \"tls.key\": \"$( cat {{ .Values.global.domain }}-agent-ca-key.pem | base64 | tr -d '\n' )\" }}" > /dev/null
-              {{- end }}
-              # Suppress globbing so we can interpolate the $NAMESPACE environment variable
-              # and use * at the start of the dns name when setting -additional-dnsname.
-              set -o noglob
```

Comment:
Do we still need the `set -o noglob`?

Everything seems to be working even if I use 
```
    serverAdditionalDNSSANs: ["*consul"]
```

But just wondering why that's not causing issues.

Reply:
That was deleted by mistake. Not sure why that isn't failing honestly? I should add it back along with the comment but maybe we did not need it in the first place? 🤷

Reply:
It looks like `sh` doing globbing but maybe the flag parsing is ignoring stuff somehow? I think we may as well keep it to be safe.

```shell-session
$ ls consul*
consul-agent-ca-key.pem  consul-agent-ca.pem

consul:
config  data    tls

$ echo consul*
consul consul-agent-ca-key.pem consul-agent-ca.pem

$ echo consul tls cert create -server -additional-dnsname consul*
consul tls cert create -server -additional-dnsname consul consul-agent-ca-key.pem consul-agent-ca.pem

$ consul tls cert create -server -additional-dnsname consul* && openssl x509 -in dc1-server-consul-0.pem -noout -text
...
            X509v3 Subject Alternative Name:
                DNS:consul, DNS:server.dc1.consul, DNS:localhost, IP Address:127.0.0.1
```

Reply:
yup. ive restored it.